extensions-core/parquet-extensions/example/compat/nested_array_struct.json

```diff
@@ -45,7 +45,8 @@
           },
           "dimensionsSpec": {
             "dimensions": [
-              "i32_dec"
+              "i32_dec",
+              "primitive"
```

Comment:
what are these changes for?

Reply:
Some of our existing tests (such as CompatParquetInputTest#testReadNestedArrayStruct) read parquet using this configs. Reading is done via the RecordReader which used to read more than needed before this PR. For example, in CompatParquetInputTest#testReadNestedArrayStruct we expect the InputRow we read to have `primitive` column. However, the spec (before this change) does not need/require/consume the `primitive` column (`primitive` is not in the metric, dimension, transform, etc). To fix the test, we can either make it so that the spec requires the `primitive` (i.e. adding `primitive` to dimensionSpec) or remove the assert check on `primitive` column. Basically, our Reader is smarter and only read what is needed breaking some existing tests.